### PR TITLE
fix #3610 media tracking fixes

### LINKED ISSUE #3610: Fix double-counting of media tracking events for new editor
Media events are currently counted twice for the new editor (or would be once launched), once in `EditPostActivity` ([here](https://github.com/wordpress-mobile/WordPress-Android/blob/398edd542e8ac72eb23385651ee25691c99de490/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L1240-L1322)) and once again in the editor fragment itself ([here](https://github.com/wordpress-mobile/WordPress-Editor-Android/blob/10a08faaeb4438251fa74bee2e80434c150d5667/WordPressEditor/src/main/java/org/wordpress/android/editor/EditorFragment.java#L721)).

We should change `EditPostActivity` to only register the events if the legacy editor is being used; otherwise, defer to the editor itself for tracking the events.

Must be merged against `feature/visual-editor`.

### PR DESCRIPTION
**Pull request by maxme**
fix #3610 media tracking fixes - also pulled latest changes from the editor subtree.

* I dropped support for multi selection count [here](https://github.com/wordpress-mobile/WordPress-Android/commit/e11b0acd24055f63a15cebf8cecbe6cbf4edebd1#diff-38b6aea9cfff5f3e0e4b7389b44de602R1621) - IMO, if we want to track that, we should create a new event. This also changes the number of events fired here, previously if the user selected 10 images using the multi-picker: one event would be sent - now 10 events will be sent.
* We previously considered Network URL as images coming from the WP media library, that's not really true.
* Note: We're not tracking the exact same EDITOR_DISCARDED_CHANGES event in iOS and Android https://github.com/wordpress-mobile/WordPress-iOS/blob/1445e067b2f62984940f260a9f8dbfed81b73e83/WordPress/Classes/ViewRelated/Post/WPLegacyEditPostViewController.m#L383-L389
* EDITOR_ENABLED_NEW_VERSION name is confusing, but we'll lost tracking history if we change it now.

### CODE REVIEW

**Comment by aforcier** [comment]
cc @meremagee

**Comment by aforcier** [comment]
The integration tests are failing (`MockEditorActivity` needs to implement `onTrackableEvent`:
```
$ ./gradlew :libs:editor:WordPressEditor:cAT --offline
error: MockEditorActivity is not abstract and does not override abstract method onTrackableEvent(TrackableEvent) in EditorFragmentListener
public class MockEditorActivity extends AppCompatActivity implements EditorFragmentAbstract.EditorFragmentListener {
       ^
1 error
```

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -1363,19 +1362,35 @@ public void onFailure(ApiHelper.ErrorType errorType, String errorMessage, Throwa
...
+            mediaUri = MediaUtils.downloadExternalMedia(this, mediaUri);
         }
 
-        if (imageUri == null) {
+        if (mediaUri == null) {
             return false;
         }
 
+        boolean isVideo = MediaUtils.isVideo(mediaUri.toString());
+        boolean isLocal = !URLUtil.isNetworkUrl(mediaUri.toString());
```
Comment (aforcier) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java]: This variable is unused.

**Comment by maxme** [comment]
@aforcier ready for review

**Comment by aforcier** [comment]
Looks good! :shipit:

**Pull request merged**